### ISSUE TITLE
[Bug]: TableHeader spreading `...rest` on wrong inner element

### ISSUE THREAD

**Issue by TannerS** [carbon-design-system/carbon#10191]
### Package

carbon-components-react

### Package version

7.48.0

### Description

In our code, we had a A11y issue (https://www.w3.org/TR/WCAG20-TECHS/H43) and wanted to make use of the `expandHeader`to help fix this. Going off examples online, I was trying to set the `id` of the th of our table (in thiscase it would be generated via the `TableHeader` component, and so I noticed the source code it has a ...rest prop to pass in any props not explicitly defined. But in our code,. I noticed, only our table headers that were NOT sortable had the id set, but table headers that were sortable did not. Looking at the source code in the `TableHeader`, I noticed this line set the `...rest` in the `th` for columns that are not sortable so that is why the `id` is getting set 

`https://github.com/carbon-design-system/carbon/blob/56cec535e90108731e5a6bb1e20975e4ddd3d1fe/packages/react/src/components/DataTable/TableHeader.js#L74`

However, I noticed for the columns that are sortable, the `th`, does not include the `...rest `, However, I did notice the inner button component DOES include the `...rest`, which makes me thing it was misplaced/typo 

`https://github.com/carbon-design-system/carbon/blob/56cec535e90108731e5a6bb1e20975e4ddd3d1fe/packages/react/src/components/DataTable/TableHeader.js#L116`

So I think this is a bug, (also attached photo to also help)

![image](https://user-images.githubusercontent.com/8866319/143958028-64cc3713-c687-4dea-8d63-b491d53bd88f.png)

**P.S. I do not mind creating a PR for this fix, I just wish to verify this is a bug first :)** 

### CodeSandbox example

N/A

### Steps to reproduce

I can't reproduce, but provided sources above 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by TannerS** [comment]
Typo in my screenshot, should say th not the word the twice :P

**Comment by tay1orjones** [comment]
@TannerS At a glance I would agree that the `...rest` should be placed on the `th`. There may be a reason as to why it is placed on the button, but I'm not sure. It's likely that moving the `...rest` could cause problems for folks expecting those props to be passed to the `button` and not the `th`. If so, we'd need to put this behind a flag for the v11 release.

**Comment by TannerS** [comment]
> @TannerS At a glance I would agree that the `...rest` should be placed on the `th`. There may be a reason as to why it is placed on the button, but I'm not sure. It's likely that moving the `...rest` could cause problems for folks expecting those props to be passed to the `button` and not the `th`. If so, we'd need to put this behind a flag for the v11 release.
> 
> Another thing to look into would be adding an `id` prop that is placed on the `th` to solve the a11y issue. That wouldn't be a breaking change.

Ya I can see that, diff odd to have ...rest in the th for non sortables then not have it for sortables lol, but one thing is clear, i cant pass a `id` prop down to the th element to help fix a few issues, so if there is soke breakage with moving the ...rest, can we find an alternative to set the id for the th? maybe a th id specific prop?

**Comment by joshblack** [comment]
Yeah definitely, we can make it so that the `id` prop is consistently placed on the outer `<th>` element 👍  What do you think @tay1orjones?

**Comment by TannerS** [comment]
@joshblack  if so I dont mind giving a wack at it

**Comment by tay1orjones** [comment]
@joshblack yeah let's go for it. 

@TannerS That would be awesome of you to contribute a fix! 🏅 Let me know if I can help.

**Comment by TannerS** [comment]
> @joshblack yeah let's go for it.
> 
> @TannerS That would be awesome of you to contribute a fix! 🏅 Let me know if I can help.

@tay1orjones hope this helps, I added a prop and tested it by using it ina few story book examples which I verified in the inspect tool in the browser 

https://github.com/carbon-design-system/carbon/pull/10214
